main.py

```diff
@@ -84,43 +87,50 @@ def shutdown(self):
         If you have some cleanup procedure, insert it here.
         """
 
-    @rest('/v2/evc/', methods=['GET'])
+    @rest("/v2/evc/", methods=["GET"])
     def list_circuits(self):
         """Endpoint to return circuits stored.
 
         If archived is set to True return all circuits, else only the ones
         not archived.
         """
-        log.debug('list_circuits /v2/evc')
-        archived = request.args.get('archived', False)
+        log.debug("list_circuits /v2/evc")
+        archived = request.args.get("archived", False)
         circuits = self.storehouse.get_data()
         if not circuits:
             return jsonify({}), 200
         if archived:
             return jsonify(circuits), 200
-        return jsonify({circuit_id: circuit
-                        for circuit_id, circuit in circuits.items()
-                        if not circuit.get('archived', False)}), 200
-
-    @rest('/v2/evc/<circuit_id>', methods=['GET'])
+        return (
+            jsonify(
+                {
+                    circuit_id: circuit
+                    for circuit_id, circuit in circuits.items()
+                    if not circuit.get("archived", False)
+                }
+            ),
+            200,
+        )
+
+    @rest("/v2/evc/<circuit_id>", methods=["GET"])
     def get_circuit(self, circuit_id):
         """Endpoint to return a circuit based on id."""
-        log.debug('get_circuit /v2/evc/%s', circuit_id)
+        log.debug("get_circuit /v2/evc/%s", circuit_id)
         circuits = self.storehouse.get_data()
 
         try:
             result = circuits[circuit_id]
         except KeyError:
-            result = f'circuit_id {circuit_id} not found'
-            log.debug('get_circuit result %s %s', result, 404)
-            raise NotFound(result)
-
+            result = f"circuit_id {circuit_id} not found"
+            log.debug("get_circuit result %s %s", result, 404)
+            raise BadRequest(result) from KeyError
         status = 200
-        log.debug('get_circuit result %s %s', result, status)
+        log.debug("get_circuit result %s %s", result, status)
         return jsonify(result), status
 
-    @rest('/v2/evc/', methods=['POST'])
-    def create_circuit(self):
+    @rest("/v2/evc/", methods=["POST"])
```

Comment:
The `description` attr of the HTTP error now being a dict, might need to be adapted accordingly in the UI to render it nicely https://github.com/ajoaoff/mef_eline/blob/oas-validation/ui/k-toolbar/main.kytos#L111, otherwise it'll dump the json object, it might be worth mapping a requirement or issue to align with Rogerio, wdyt? 
![2021-11-29-190808_1326x767_scrot](https://user-images.githubusercontent.com/1010796/144056328-09b8ab6f-2f03-4ee7-a6a3-64e60470d7bb.png)

Reply:
I'm really not sure if we should return this `dict` on error, it is too verbose. What do you think if we just return a`string` message and log the full dictionary?

Reply:
Indeed, great idea summarizing the relevant parts of the error in the `description` string, and then if ever the front-end or any other client needs more context we can keep adding/refining additional fields, @rmotitsuki do you have any plans for richer error reporting/identifying in the front-end? Does the current string approach seems good for the foreseeable future?

Reply:
For now the description should be enough, if it identifies the field with error.
We can open an issue to refine the UI requirements for the EVC form. The UI should block the request and warn the user if the format for the field is not met.

Reply:
Sounds good, Rogerio, with the latest commits that @ajoaoff pushed the errors include the fields

```
{
    "code": 400,
    "description": "'1' is too short for field uni_a/interface_id.",
    "name": "Bad Request"
}
```

```
{
    "code": 400,
    "description": "True is not of type string for field uni_a/interface_id.",
    "name": "Bad Request"
}
```